Source/Core/Core/ConfigManager.cpp

```diff
@@ -131,6 +131,10 @@ void SConfig::SaveGeneralSettings(IniFile& ini)
 #endif
   general->Set("GDBPort", iGDBPort);
 #endif
+
+#ifdef USE_DISCORD_PRESENCE
+  general->Set("UseDiscordPresence", bUseDiscordPresence);
```

Comment:
Please use the new config system for this, not SConfig. This doesn't really belong in Core and we are currently in the process of removing things from SConfig.

Reply:
Sorry, I'm feel a bit confused by this comment.
Are those examples of this new config system you are talking about? https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/Config/MainSettings.cpp
I also don't see any those outside of Core, so I don't where to write to.

Reply:
Yeah. You include `Common/Config/Config.h`, add a ConfigInfo for your setting and use Config::Set/Config::Get respectively to set/get the setting value.

I'd add the `ConfigInfo<bool>` to UICommon/DiscordPresence.

Reply:
Okay, but one thing. I think you forgot to tell me how to save the setting into the ini file. After closing dolphin, the setting goes back what it was before.